metricbeat/module/vsphere/resourcepool/resourcepool.go

```diff
@@ -0,0 +1,150 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package resourcepool
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/property"
+	"github.com/vmware/govmomi/view"
+	"github.com/vmware/govmomi/vim25/mo"
+
+	"github.com/elastic/beats/v7/metricbeat/mb"
+	"github.com/elastic/beats/v7/metricbeat/module/vsphere"
+)
+
+// init registers the MetricSet with the central registry as soon as the program
+// starts. The New function will be called later to instantiate an instance of
+// the MetricSet for each resourcepool is defined in the module's configuration. After the
+// MetricSet has been created, Fetch will be called periodically.
+func init() {
+	mb.Registry.MustAddMetricSet("vsphere", "resourcepool", New,
+		mb.WithHostParser(vsphere.HostParser),
+		mb.DefaultMetricSet(),
+	)
+}
+
+// ResourcePoolMetricSet type defines all fields of the MetricSet.
+type ResourcePoolMetricSet struct {
+	*vsphere.MetricSet
+}
+
+// New creates a new instance of the MetricSet. New is responsible for unpacking
+// any MetricSet specific configuration options if there are any.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	ms, err := vsphere.NewMetricSet(base)
+	if err != nil {
+		return nil, err
+	}
+	return &ResourcePoolMetricSet{ms}, nil
+}
+
+// metricData holds performance metrics values.
+type metricData struct {
+	assetNames assetNames
+}
+
+type assetNames struct {
+	outputVmNames []string
+}
+
+// Fetch methods implements the data gathering and data conversion to the right
+// format. It publishes the event which is then forwarded to the output. In case
+// of an error set the Error field of mb.Event or simply call report.Error().
+func (m *ResourcePoolMetricSet) Fetch(ctx context.Context, reporter mb.ReporterV2) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
```

Comment:
@gpop63 what you are saying is correct. we are just following already implemented code and try not to change any coding standards already followed.

Reply:
@gpop63 you are right. @kush-elastic is it like this all metricsets?

Reply:
Yes

Reply:
@kush-elastic I agree with @gpop63 here. It definitely is redundant. If you notice, the `defer cancel()` isn't providing any additional value because the context isn't canceled explicitly. Think of it way this way, since it will only be called when Fetch returns, would it have any affect on how this function is stopped ? If the answer is No, simply remove it. 
Also, I'm of the opinion that it's not really a coding standard that says you have to create a new child context at the start of the function. If it doesn't fit, we should omit it.

Reply:
@kush-elastic @niraj-elastic : Can we please make a note of these suggestions which would be applicable across metricsets. So that we can make the change across metricsets. 
There was another one suggested in datastore_cluster PR.

We can add this to the future_tasklist.